### PR TITLE
[MM-56653] Improve license loading errors

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
If loading the license on startup fails, the only error message printed is `"No valid enterprise license found"`. This PR improves the error handling of license errors by returning idiomatic an `error` in `ValidateLicense`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56653

#### Release Note
```release-note
Improve license loading errors
```

### CODE REVIEW

File: server/channels/utils/license.go [hunk 1]
```diff
@@ -32,33 +33,32 @@ func init() {
...
 
-func (l *LicenseValidatorImpl) ValidateLicense(signed []byte) (bool, string) {
+func (l *LicenseValidatorImpl) ValidateLicense(signed []byte) (string, error) {
 	decoded := make([]byte, base64.StdEncoding.DecodedLen(len(signed)))
 
 	_, err := base64.StdEncoding.Decode(decoded, signed)
 	if err != nil {
-		mlog.Error("Encountered error decoding license", mlog.Err(err))
-		return false, ""
+		return "", fmt.Errorf("Encountered error decoding license: %w", err)
```
Comment (isacikgoz) [inline server/channels/utils/license.go:61]: nit: Shall we use lower case on errors?

**Review by isacikgoz — APPROVED** [review]
LGTM, we can maybe use lowercase errors, there are other places we don't mind that but since we are here I think capitalized errors are not necessary. Leaving it to you 0/5

File: server/channels/utils/license.go (hunk 1 above)
Comment (hanzei) [reply server/channels/utils/license.go]: Good point, updated :+1:

**Review by nickmisasi — APPROVED** [review]

**Comment by hanzei** [comment]
@isacikgoz @nickmisasi Heads up that I added some more changes via [`26a765e` (#26050)](https://github.com/mattermost/mattermost/pull/26050/commits/26a765e4f699d1c827386ffa4e13f72b047a0d22). `ValidateAndSetLicenseBytes` now returns a proper `error` as previously they would not get logged.

**Review by lindy65 — APPROVED** [review]
Thanks @hanzei - smoke tested adding and removing licences and checking that the various features associated with those licences were enabled/disabled appropriately. Looks good to me 👍

**Pull request merged**